### ISSUE TITLE
Realm object observing?

### ISSUE THREAD

**Issue by winstondu** [arturdev/Unrealm#38]
Hey Arturo!

This seems like an amazing library. However, one useful paradigm we use (with RxRealm) is the ability to observe changes on an object in the realm db and register a callback.

This function is the `public func observe<T>(on queue: DispatchQueue? = nil, _ block: @escaping (RealmSwift.ObjectChange<T>) -> Void) -> RealmSwift.NotificationToken where T : RealmSwift.Object`.

Does the fact that the objects are no longer "live" make this no longer possible?

**Comment by arturdev** [comment]
Hi @winstondu. Unfortunately yes. The observing which you're looking for is done using ObjectiveC's KVO mechanism... And as you might already know,  Swift native types don't support KVO

**Comment by winstondu** [comment]
I see. But in your library, any write to the `Realmable` object will still be enclosed in a `Realm.write` no? 

Is there no way to know whether the `Realmable` is updated (even if we don't know which properties are)?

**Comment by arturdev** [comment]
Yes, that is possible, but as you already mentioned, you will not know which property exactly was changed.

**Comment by winstondu** [comment]
@arturdev , what would be the function when using this library to make change observations?
